streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/core/master/scheduler/strategy/impl/PipelineFirstStrategy.java

```diff
@@ -0,0 +1,214 @@
+package org.ray.streaming.runtime.core.master.scheduler.strategy.impl;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.ray.streaming.runtime.config.types.SlotAssignStrategyType;
+import org.ray.streaming.runtime.core.graph.executiongraph.ExecutionGraph;
+import org.ray.streaming.runtime.core.graph.executiongraph.ExecutionJobVertex;
+import org.ray.streaming.runtime.core.graph.executiongraph.ExecutionVertex;
+import org.ray.streaming.runtime.core.master.scheduler.strategy.SlotAssignStrategy;
+import org.ray.streaming.runtime.core.resource.Container;
+import org.ray.streaming.runtime.core.resource.ContainerID;
+import org.ray.streaming.runtime.core.resource.Resources;
+import org.ray.streaming.runtime.core.resource.Slot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
```

Comment:
This is the most complex class in this PR, therefore a comprehensive UT is required.

Reply:
Ok, added.